src/main/java/vazkii/patchouli/api/PatchouliAPI.java

```diff
@@ -265,6 +265,20 @@
 		 * Gets an IStateMatcher that accepts anything.
 		 */
 		IStateMatcher anyMatcher();
+
+		// ================================================================================================
+		// Macros
+		// ================================================================================================
+
+		/**
+		 * Registers a command macro handler under the given names
+		 */
+		void registerMacro(ICommandProcessor handler, String... names);
+
+		/**
+		 * Registers a function macro handler under the given names
+		 */
+		void registerMacro(IFunctionProcessor handler, String... names);
```

Comment:
throw on duplicate registration and document that

Reply:
so I should throw not only when it's not registered but if one name is duplicated?